addons/wardrobe/functions/fnc_replace.sqf

```diff
@@ -0,0 +1,95 @@
+#include "../script_component.hpp"
+/*
+ * Author: OverlordZorn
+ * Ace Action Statement - Removes the modifiableItem and replaces it with the target item
+ *
+ * Arguments:
+ * 0: Action target <OBJECT>
+ * 1: Action player <OBJECT>
+ * 2: Action params <ARRAY>
+ * - 0: Current variant <CONFIG>
+ * - 0: Desired variant <CONFIG>
+ * 3: Replace Now? <BOOL> (default: false)
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * _this call ace_wardrobe_fnc_replace
+ *
+ * Public: No
+ */
+
+params ["_target", "_unit", "_actionParams", ["_replaceNow", false, [true]]];
+_actionParams params ["_cfgOrigin", "_cfgTarget"];
+
+// Duration of the "animation"
+private _duration = getNumber (configFile >> QUOTE(ADDON) >> configName _cfgTarget >> "duration");
+if (_replaceNow) then { _duration = 0; }; // needed for cba context menu - avoid potential duplications and such
+
+// Replace the Main Item.
+private _additionalParams = "";
+private _typeNumber = getNumber (_cfgOrigin >> "ItemInfo" >> "type");
+private _replaceCode = switch ( _typeNumber ) do {
+    case TYPE_HEADGEAR: { _additionalParams = "HEADGEAR"; FUNC(replaceOther) };
+    case TYPE_UNIFORM:  { _additionalParams = "UNIFORM";  FUNC(replaceContainer)  };
+    case TYPE_VEST:     { _additionalParams = "VEST";     FUNC(replaceContainer)  };
+    case TYPE_BACKPACK: { _additionalParams = "BACKPACK"; FUNC(replaceContainer)  };
+    default {
+        // CfgGlasses items do not have a ItemInfo Subclass and therefore, not TypeNumber.
+        switch (true) do {
+            case (isClass (configFile >> "CfgGlasses" >> configName _cfgOrigin)): { _additionalParams = "FACEWEAR"; FUNC(replaceOther) };
+            default { false };
+        };
+    };
+};
+
+if (_replaceCode isEqualType false) exitWith { ERROR_2("typeNumber undefined: %1 - %2",_typeNumber,configName _cfgOrigin); };
+[ _replaceCode, [_unit, _cfgOrigin, _cfgTarget, _additionalParams ], _duration] call CBA_fnc_waitAndExecute;
+
+//// Handle components
+[_cfgOrigin, _cfgTarget] call FUNC(compareComponents) params ["_missing", "_surplus"];
+
+// Add Surplus
+{
+    if (configName _cfgTarget isNotEqualTo _x) then {
+        if ( isClass (configFile >> "CfgGlasses" >> _x) && { goggles _unit isEqualTo "" } ) then {
+            _unit addGoggles _x;
+        } else {
+            [_unit, _x, true] call CBA_fnc_addItem;
```

Comment:
Side note: `CBA_fnc_addItem` is currently bugged, but I'll be making a PR to fix it.